src/Entity/ReactionRuleStorage.php

```diff
@@ -84,6 +85,7 @@ protected function getRegisteredEvents() {
     $events = [];
     foreach ($this->loadMultiple() as $rules_config) {
       $event = $rules_config->getEvent();
+      $event = RulesEventManager::rulesGetEventBaseName($event);
```

Comment:
You need to get the rules event manager service here. If not done yet, inject it first.